### PR TITLE
Update use cases in Developer Guide

### LINKED ISSUE #25: Update Developer Guide (Use Cases)
# TODO
Give use cases (textual form) for a few representative user stories that need multiple steps to complete. 
_e.g. Adding a tag to a person (assume the user needs to find the person first)_
See sample [here](https://se-education.org/addressbook-level3/DeveloperGuide.html#use-cases).

## Things to note (and/or avoid)

- Important use cases missing (a use case is important if it involves a user interaction that is worthy of documenting e.g., it has multiple extensions -- this is not the same as the feature being important)
- Formatting/notational errors
- Incorrect step numbering
- Unnecessary UI details mentioned
- Missing/unnecessary steps
- Missing extensions

### PR DESCRIPTION
**Pull request by melissaharijanto**

### CODE REVIEW

**Comment by melissaharijanto** [comment]
If there is anything I missed, please let me know and I'll add it to the use cases!

**Comment by melissaharijanto** [comment]
Also please be critical with the extension numbers because I am unsure about a few of them! :D

**Review by janelleljt — APPROVED** [review]
Looks good!

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -283,32 +283,155 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+2. System lists all orders.
 
-      Use case resumes at step 2.
+    Use case ends.
 
-*{More to be added}*
+**Extensions**
+
+- 2a. The list of orders is empty.
+    - 2ai. System informs the user that the list is empty.
```
Comment (donljh) [inline docs/DeveloperGuide.md]: I could be mistaken, but I think the numbering here should be **2a** then **2a1**, i.e, no use of roman numerals for sublisting.

Do check all the other extensions as well!

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -283,32 +283,155 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+  - 1ai. System informs user of the non-existent item.
+    
+    Use case ends.
+
+- 2a. The list of inventory items is empty.
+    - 2ai. System informs user that the list of items is empty.
+
+      Use case ends.
+    
+**MSS**
```
Comment (donljh) [inline docs/DeveloperGuide.md]: I think you might accidentally added this "MSS" text here.

Do check the rest of the document for similar mistakes

**Review by donljh — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (melissaharijanto) [reply docs/DeveloperGuide.md]: Ah, it's my bad! Fixed all of them :D thank you for pointing it out!

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (melissaharijanto) [reply docs/DeveloperGuide.md]: Thank you for pointing this out!

**Review by donljh — APPROVED** [review]
Good work!

**Pull request merged**